### PR TITLE
Upsert implemented

### PR DESCRIPTION
**Pull request by albertossilva**

### CODE REVIEW

**Review by f3liperamos — APPROVED** [review]
Why does sequelize.js needs an model *AND* a tableName? :(

:+1: For upsert and resolves.

File: src/BaseRepository.js [hunk 1]
```diff
@@ -90,6 +90,11 @@ class BaseRepository {
       .then(() => entityInstance)
   }
 
+  upsert (insertValues, updateValues, where) {
```
Comment (danielbdias) [inline src/BaseRepository.js:93]: Cool!

File: src/BaseRepository.js [hunk 2]
```diff
@@ -6,29 +6,29 @@ class BaseRepository {
         sequelizeModel,
         mapper,
         sequelize: sequelizeModel && sequelizeModel.sequelize,
-        queryInterface: sequelizeModel && sequelizeModel.sequelize.getQueryInterface()
+        queryInterface: sequelizeModel && sequelizeModel.sequelize.getQueryInterface(),
```
Comment (danielbdias) [inline src/BaseRepository.js:9]: Just wondering, this will not open or keep an active connection with the database, right?

Comment (albertossilva) [reply src/BaseRepository.js:9]: no... sequelize controls the connection pooling.

**Pull request merged**